witness: Internal Server Error when trying to query oobi for delegated multisig AID

Currently, we are experiencing internal server errors in witnesses for a multisig AID that has been delegated from another multisig AID created in keripy. The error we are experiencing is this:

```
2024-04-17 12:42:59 [FALCON] [ERROR] GET /oobi/<multisig-prefix>/witness/<witness-prefix> => Traceback (most recent call last):
--
File "falcon/app.py", line 365, in falcon.app.App.__call__
File "/keripy/src/keri/end/ending.py", line 594, in on_get
msgs = hab.replyToOobi(aid=aid, role=role, eids=eids)
File "/keripy/src/keri/app/habbing.py", line 2007, in replyToOobi
return self.replyEndRole(cid=aid, role=role, eids=eids)
File "/keripy/src/keri/app/habbing.py", line 1962, in replyEndRole
msgs.extend(self.replay(cid))
File "/keripy/src/keri/app/habbing.py", line 1546, in replay
for msg in self.db.cloneDelegation(kever=kever):
File "/keripy/src/keri/db/basing.py", line 1307, in cloneDelegation
dkever = self.kevers[kever.delegator]
File "/keripy/src/keri/db/basing.py", line 68, in __getitem__
raise ex  # reraise KeyError
File "/keripy/src/keri/db/basing.py", line 63, in __getitem__
return super(dbdict, self).__getitem__(k)
KeyError: '<delegator-prefix>'
```

The impliciation of this is that no one can resolve the OOBI of our multisig AID. We have not been able to create a consistent reproduction. There is a related issue in keria that is also having problems with the same AID: https://github.com/WebOfTrust/keria/issues/231.

Does anyone have any clue of what the cause might be?

We are using keripy version: 1.1.10

--- a/.github/workflows/python-app-ci.yml
+++ b/.github/workflows/python-app-ci.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ macos-latest, ubuntu-latest ]
+        os: [ macos-13, ubuntu-latest ]
 
     steps:
       - uses: actions/checkout@v3

--- a/src/keri/db/basing.py
+++ b/src/keri/db/basing.py
@@ -1261,6 +1261,9 @@ class Baser(dbing.LMDBer):
             atc.extend(coring.Counter(code=coring.CtrDex.SealSourceCouples,
                                       count=1).qb64b)
             atc.extend(couple)
+        elif self.kevers[pre].delegated:
+            if coring.SerderKERI(raw=raw).estive:
+                raise kering.MissingEntryError("Missing delegator anchor seal for dig={}.".format(dig))
 
         # add trans receipts quadruples to attachments
         if quads := self.getVrcs(key=dgkey):

--- a/src/keri/end/ending.py
+++ b/src/keri/end/ending.py
@@ -577,6 +577,10 @@ class OOBIEnd:
             rep.status = falcon.HTTP_NOT_FOUND
             return
 
+        if kever.delegated and kever.delegator not in self.hby.kevers:
+            rep.status = falcon.HTTP_NOT_FOUND
+            return
+
         owits = oset(kever.wits)
         if kever.prefixer.qb64 in self.hby.prefixes:  # One of our identifiers
             hab = self.hby.habs[kever.prefixer.qb64]

--- a/tests/end/test_ending.py
+++ b/tests/end/test_ending.py
@@ -434,6 +434,49 @@ def test_get_oobi():
         assert serder.ked['t'] == coring.Ilks.icp
         assert serder.ked['i'] == "EOaICQwhOy3wMwecjAuHQTbv_Cmuu1azTMnHi4QtUmEU"
 
+    delname = "delegator"
+    with habbing.openHby(name=name, base=base, salt=salt) as hby, \
+            habbing.openHby(name=delname, base=base, salt=salt) as delhby:
+        delhab = delhby.makeHab(name=delname)
+        hab = hby.makeHab(name=name, delpre=delhab.pre)
+
+        assert hab.pre == "EPERMS4wKU7ejhCdhI2qQR8snEx1cislR9C9bSEs0kS5"
+        assert hab.kever.delegator == delhab.pre
+
+        msgs.extend(hab.makeEndRole(eid=hab.pre,
+                                    role=kering.Roles.controller,
+                                    stamp=help.nowIso8601()))
+
+        msgs.extend(hab.makeLocScheme(url='http://127.0.0.1:5555',
+                                      scheme=kering.Schemes.http,
+                                      stamp=help.nowIso8601()))
+        hab.psr.parse(ims=msgs)
+
+        # must do it here to inject into Falcon endpoint resource instances
+        tymist = tyming.Tymist(tyme=0.0)
+
+        app = falcon.App()  # falcon.App instances are callable WSGI apps
+        ending.loadEnds(app, tymth=tymist.tymen(), hby=hby, default=hab.pre)
+
+        client = testing.TestClient(app=app)
+
+        # This should fail with 404 because we haven't been approved yet so we don't exist
+        rep = client.simulate_get('/oobi', )
+        assert rep.status == falcon.HTTP_NOT_FOUND
+
+        # Approve the delegation manually
+        delhab.interact(data=[dict(i=hab.pre, s="0", d=hab.pre)])
+        for msg in delhab.db.clonePreIter(pre=delhab.pre, fn=0):
+            hab.psr.parse(ims=msg)
+
+        rep = client.simulate_get('/oobi', )
+        assert rep.status == falcon.HTTP_OK
+
+        # We'll get the delegator first
+        serder = serdering.SerderKERI(raw=rep.text.encode("utf-8"))
+        assert serder.ked['t'] == coring.Ilks.icp
+        assert serder.ked['i'] == "EKL3to0Q059vtxKi7wWmaNFJ3NKE1nQsOPasRXqPzpjS"
+
     """Done Test"""
 
 

